packages/commonwealth/client/scripts/hooks/api/useTopics.tsx

```diff
@@ -0,0 +1,294 @@
+import React, { useState } from 'react';
+import { Topic } from 'models';
+import { useMutation, useQueryClient, useQuery } from '@tanstack/react-query';
+import $ from 'jquery';
+import app from 'state';
+import queryKeys from './queryKeys';
+
+const useTopics = () => {
+  // auth state to come from zustand, localstorage is just for demo purposes
+  const JWT = localStorage.getItem('jwt');
+
+  const queryClient = useQueryClient();
+  const [topicsForChainId, setTopicsForChainId] = useState<string>();
+
+  const fetchTopicsQuery = useQuery({
+    queryKey: [queryKeys.TOPICS_FOR_CHAIN(topicsForChainId)],
+    queryFn: async () => {
+      return queryClient.getQueryData([
+        queryKeys.TOPICS_FOR_CHAIN(topicsForChainId),
+      ]);
+    },
+    enabled: !!topicsForChainId,
+    staleTime: 60_000, // cache for 60 seconds
+  });
+
+  const addTopicQuery = useMutation({
+    onMutate: async ({
+      name,
+      description,
+      telegram,
+      featuredInSidebar,
+      featuredInNewPost,
+      tokenThreshold,
+      defaultOffchainTemplate,
+    }: {
+      name: string;
+      description: string;
+      telegram: string;
+      featuredInSidebar: boolean;
+      featuredInNewPost: boolean;
+      tokenThreshold: string;
+      defaultOffchainTemplate: string;
+    }) => {
+      // Cancel any outgoing refetches ,(so they don't overwrite
+      // our optimistic update)
+      await queryClient.cancelQueries({
+        queryKey: [queryKeys.TOPICS_FOR_CHAIN(topicsForChainId)],
+      });
+
+      // fetch old topics from react query store
+      const oldTopics = queryClient.getQueryData([
+        queryKeys.TOPICS_FOR_CHAIN(topicsForChainId),
+      ]);
+
+      // update optimistic value
+      queryClient.setQueryData(
+        [queryKeys.TOPICS_FOR_CHAIN(topicsForChainId)],
+        (t: Topic[]) => {
+          const updatedTopics = [...t];
+
+          const newTopic = new Topic({
+            id: -1, // -1 to indicate a new thread
+            name,
```

Comment:
`-1` here indicates a new optimistic resources. To keep things simple i used `-1` but ideally we would need to figure out a way to add unique ids/constraints to optimistic data.